packages/browser-repl/src/components/utils/inspect.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const utilInspect = require('util').inspect;
+const utilInspectCustom = utilInspect.custom || 'inspect';
+
+const formatBsonType = (value): any => ({
+  [utilInspectCustom](): string {
+    return `${value._bsontype}(${(JSON.stringify(value))})`;
+  }
+});
+
+function isBsonType(value): any {
+  return value && value._bsontype;
+}
+
+function isObject(value): any {
+  return value && typeof value === 'object' && !Array.isArray(value);
+}
+
+function formatProperty(value): any {
+  if (isObject(value) && isBsonType(value)) {
```

Comment:
so i think our [js-bson](https://github.com/mongodb/js-bson) package will format the value if it's a bson type, and otherwise leave it as is. and if you end up using it, you might not need this check.